indev/xkb.c

```diff
@@ -79,15 +79,15 @@ bool xkb_init_state(xkb_drv_state_t *state) {
  * @param state XKB driver state to use
  */
 void xkb_deinit_state(xkb_drv_state_t *state) {
-  if (state->keymap) {
-    xkb_keymap_unref(state->keymap);
-    state->keymap = NULL;
-  }
-
   if (state->state) {
     xkb_state_unref(state->state);
     state->state = NULL;
   }
+
+  if (state->keymap) {
+    xkb_keymap_unref(state->keymap);
+    state->keymap = NULL;
+  }
```

Comment:
This shouldn't really matter much but the state references the keymap. So it seems more logical to free the memory in this order.